src/api/Controllers/BoreholeFileController.cs

```diff
@@ -0,0 +1,135 @@
+﻿using BDMS.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.ComponentModel.DataAnnotations;
+
+namespace BDMS.Controllers;
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class BoreholeFileController : ControllerBase
+{
+    private readonly BdmsContext context;
+    private readonly BoreholeFileUploadService boreholeFileUploadService;
+    private readonly ILogger logger;
+
+    public BoreholeFileController(BdmsContext context, ILogger<BoreholeFileController> logger, BoreholeFileUploadService storageService)
+        : base()
+    {
+        this.logger = logger;
+        boreholeFileUploadService = storageService;
+        this.context = context;
+    }
+
+    /// <summary>
+    /// Uploads a file to the cloud storage and links it to the borehole.
+    /// </summary>
+    /// <param name="file">The file to upload and link to the <see cref="Borehole"/>.</param>
+    /// <param name="boreholeId">The <see cref="Borehole.Id"/> to link the uploaded <paramref name="file"/> to.</param>
+    [HttpPost("upload")]
+    public async Task<ActionResult> Upload(IFormFile file, [Range(1, int.MaxValue)] int boreholeId)
+    {
+        if (file == null || file.Length == 0) return BadRequest("No File provided.");
+        if (boreholeId == 0) return BadRequest("No BoreholeId provided.");
```

Comment:
PP: Exakten Parameter-Namen in Fehlermeldung ausgeben
```suggestion
        if (boreholeId == 0) return BadRequest("No boreholeId provided.");
```